test/unit/alignment/pairwise/execution/execution_handler_template.hpp

```diff
@@ -12,93 +12,136 @@
 
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
 #include <seqan3/range/views/view_all.hpp>
+#include <seqan3/range/views/zip.hpp>
 #include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/std/iterator>
+#include <seqan3/std/algorithm>
+#include <seqan3/std/ranges>
 
 using namespace seqan3;
 
 template <typename T>
-class execution_handler : public ::testing::Test
-{};
-
-TYPED_TEST_CASE_P(execution_handler);
-
-TYPED_TEST_P(execution_handler, execute_w_lvalue)
+struct execution_handler : public ::testing::Test
 {
-    constexpr size_t SIZE = 10000;
-    std::vector<std::pair<size_t, size_t>> buffer;
-    buffer.resize(SIZE);
+    static constexpr size_t total_size = 10000;
 
-    TypeParam exec_handler{};
+    void SetUp()
+    {
+        for (unsigned i = 0; i < total_size; ++i)
+        {
+            sequence_collection1.push_back(test::generate_sequence<dna4>(100, 20, i));
+            sequence_collection2.push_back(test::generate_sequence<dna4>(100, 20, i + total_size));
+        }
+    }
 
-    auto callable = [](size_t const idx, auto && rng1, auto && rng2)
+    template <typename buffer_t>
+    void check_result(buffer_t const & buffer) const
     {
-        return std::pair{idx, rng1.size() + rng2.size()};
-    };
+        for (unsigned i = 0; i < total_size; ++i)
+        {
+            EXPECT_EQ(buffer[i].first, i) << "Position: " << i;
+            EXPECT_EQ(buffer[i].second,
+                      sequence_collection1[i].size() + sequence_collection2[i].size()) << "Position: " << i;
+        }
+    }
 
-    std::vector<dna4_vector> set1;
-    std::vector<dna4_vector> set2;
+    std::vector<dna4_vector> sequence_collection1{};
+    std::vector<dna4_vector> sequence_collection2{};
+};
+
+auto simulate_alignment = [](size_t const idx, auto && rng1, auto && rng2)
+{
+    return std::pair{idx, rng1.size() + rng2.size()};
+};
 
-    for (unsigned i = 0; i < SIZE; ++i)
+auto simulate_alignment_with_range = [] (auto indexed_sequence_pairs)
+{
+    std::vector<std::pair<size_t, size_t>> results{};
+    for (auto && [sequence_pair, idx] : indexed_sequence_pairs)
     {
-        set1.push_back(test::generate_sequence<dna4>(100, 20, i));
-        set2.push_back(test::generate_sequence<dna4>(100, 20, i + SIZE));
+        results.emplace_back(idx, std::get<0>(sequence_pair).size() + std::get<1>(sequence_pair).size());
     }
 
-    size_t pos = 0;
+    return results;
+};
+
+TYPED_TEST_CASE_P(execution_handler);
+
+TYPED_TEST_P(execution_handler, execute_w_lvalue)
```

Comment:
what is a `w_lvalue`?

EDIT: Do you mean `with`? :laughing: please be expressive in the tests :innocent:

Reply:
yes it is supposed to read `with`.